pkg/controller/cluster/cluster_controller.go

```diff
@@ -201,15 +224,20 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, nil
 
 	} else if utils.Contains(instance.ObjectMeta.Finalizers, clusterFinalizer) {
-		// our finalizer is present, so delete cluster first
-		err := k.DeleteCluster(instance.Spec.KopsConfig)
+
+		//check if cluster still exists
+		err := k.GetCluster(instance.Spec.KopsConfig)
 		if err != nil {
-			// FIXME - Ensure that delete implementation is idempotent and safe to invoke multiple times.
-			// If we call delete and the cluster is not present it will cause error and it will keep erroring out
-			//return reconcile.Result{}, err
+			reqLogger.Info("Cluster is already deleted...")
```

Comment:
```suggestion
			reqLogger.WithValues("error", err).Info("Cluster is already deleted...")
```
Or relevant syntax to this logger